Guide/Greedy/Interval Scheduling/README.md

```diff
@@ -0,0 +1,181 @@
+
+### Input Format
+- A set _S_ containing n rquests in the form of _(s, f)_ where _s_ is the start time and _f_ is the finish time of a request.
+
+### Constraints
+- No two intervals _(s, f)_ may overlap within a solution, and the solution must be the most optimal soltuion of _S_, i.e. maximum number of requests in the solution.
+
+### Output Format
+- a list of the intervals existing in the optimal solution.
+
+## Algorithm
+### Overview
+For our interval scheduling algorithm we will re-state the problem in a more precise way:
+
+> Given a set _R_ = {i<sub>1</sub>...,i<sub>n</sub>} of requests where each request _i_ is associated with an interval [s(i), f(i)), find a best schedule, i.e. a subset _A_ of requests from _R_ such that:
+
+> 1. _A_ is _compatible_, i.e. _A_ contains no overlapping requests and 
+
+> 2. _A_ is _optimal_, i.e. there is no compatible subset of _R_ which has more requests than _A_.
+
+
+### Pseudo Code
+
+## Version 1
+
+```python
+
+    def interval_scheduling(set_of_requests):
+      1. While set_of_requests is not empty
+        a. select a request x from set_of_requests with earliest finish time
+        b. add x it to the solution set
+        c. remove all all sets incompatible with x from set_of_requests including x
+      2. return the solution set
+```
+
+## Version 2
+
+```python
+    def interval_scheduling(set_of_requests):
+        1. Sort set_of_requests in order of finishing time.
+        2. i = 0, n = len(set_of_requests)
+        3. f := − 1
+        4. while i <= n:
+            a. if set_of_requests[i]'s finish time is greater than or equal to f:
+                1. append the request to the solution set
+                2. f = the requests finish time
+            b. i = i + 1
+        5. return the solution set
+
+```
+
+### Analysis
+
+## Version 1
+
+Version 1 of the solution has a worst case time complexity of O(n<sup>2</sup>) and a best case of O(n). Since the list of requests is not sorted
+the algorithm must loop over the list of requests n times for the outer while loop, and n more times for finding the minimal finish time.
+
+## Example
+
+Requests = [(6, 9), (2, 3), (1, 4), (1, 3), (4, 5)]
+
+Solution = []
+
+Step 1:
+
+    a. first request with minimum finish time is (2, 3)
+    
+    b. Solution = [(2, 3)]
+    
+    c. Requests = [(6, 9), (4, 5)]
+
+Step 2:
+
+    a. request with minimum finish time is (4, 5)
+    
+    b. Solution = [(2, 3), (4, 5)]
+    
+    c. Requests = [(6, 9)]
+
+Step 3:
+
+    a. request with minimum finsih time is (6, 9)
+    
+    b. Solution = [(2, 3), (4, 5), (6, 9)]
+    
+    c. Request = []
+
+Solution set = [(2, 3), (4, 5), (6, 9)]
+
+## Version 2
+
+Version 2 of the solution has a worst case time complexity of O(n<sup>2</sup>) and an average time complexity of O(n log n) due to the sorting of the set of request before finding the optimal solution.
```

Comment:
"request" => "requests". You also might consider switching to something like "O(n log n) from quick sorting the set of requests before finding the optimal solution." This reduces the number of connective words while also making clear that you're using quick sort. I know you specify which sorting algorithm you're using in the code, but some people read the docs first.